src/include/sof/math/trig.h

```diff
@@ -12,6 +12,14 @@
 
 #include <stdint.h>
 
+#if defined __XCC__
+#include <xtensa/config/core-isa.h>
+#if XCHAL_HAVE_HIFI3 == 1
+#define CORDIC_HIFI3
```

Comment:
why define your own rather than use the provided define?

Reply:
Umm, do yo mean use of XCHAL_HAVE_HIFI3 directly? I was thinking CORDIC_HIFI3 as generic HiFi3 compatible code enabler. While we don't have HiFi4 version done yet this could be set also for build in a HiFi4 platform due to backwards compatibility.

Reply:
Rather than going around and writting `|| XCHAL_HAVE_HIFI4` makes sens to me